### ISSUE TITLE
Postgres createDatabase ifNotExists leaks connection

### ISSUE THREAD

**Issue by mhahnenberg** [tada5hi/typeorm-extension#206]
https://github.com/tada5hi/typeorm-extension/blob/7e080bd1a095127e1c05f06dadf36fd46d0e8158/src/database/driver/postgres.ts#L67

You can see that the query to check if the database exists is told not to close the connection (with`endConnection` set to `false`), but if the result of the query indicates the database already exists we early return with `Promise.resolve()` without closing the connection.

I observed this when using `createDatabase` in a test suite. The symptom was that jest would finish running tests but hang and never exit. If I removed the `createDatabase` call it worked fine. I was able to hackily work around this by calling `createDatabase` with `ifNotExist` set to `false` and just using a `try-catch` for the error thrown by the postgres driver, e.g.:

```
try {
    await createDatabase({ ifNotExist: false, options: AppDataSource.options });
  } catch (e) {
    console.log('Error creating test database:', e);
  }
```

Not the end of the world, but would be nice not to have to do this.

**Comment by tada5hi** [comment]
@mhahnenberg thank you for reporting this issue. This is a good catch. I will fix it as soon as possible.

**Comment by tada5hi** [comment]
Can you confirm, that `2.1.1` solves your issue ?

**Comment by mhahnenberg** [comment]
> @mhahnenberg can you confirm, that `v2.1.1` solves your issue ?

Yep, just tested it locally without my workaround and jest exits cleanly now 😌 Thanks for the quick fix!

**Comment by tada5hi** [comment]
Glad to hear.
Your welcome 😇
